### ISSUE TITLE
How does this plugin work?

### ISSUE THREAD

**Issue by oddbear** [janecker/goxlr-interface#1]
Hi.

Is it possible to get some more explaination/documentation about how this app works?
It seems to run in parrallel to the StreamDeck plugin, and not instead of, is this correct (cannot find any references to the port etc.)?

I managed to run it as a electron app (I think), but I am kind of stuck now (JS is kind of not my cup of tea).

**Comment by janecker** [comment]
Hi @oddbear,

I am currently on my phone and will therefore keep it short ;). You are right, the app is supposted to run in parallel to the GoXLR app. You have to extract the streamdeck plugin from the rar you can find in your installation and set the path in the settings. I can try to commit my latest changes tomorrow. Maybe I will also try to provide a better how to, now that I know that someone is actually interested in the app.

**Comment by GinjahWolf92** [comment]
Sorry, was following this repo and never said anything but thank you for the amazing work, I am using this software and it's so helpful! :)

Kind Regards

Kieran Matheis

________________________________

**Comment by janecker** [comment]
Hi @GinjahWolf92,

wow, thanks for the feedback, would I have know would I already have committed my latest changes xD. There will definitely be an update tomorrow...

**Comment by janecker** [comment]
I just pushed my latest changes with many bugfixes and a new small how to that explains the basics of getting it running. If you are still facing any issues don't hassitate to reach out again!

**Comment by oddbear** [comment]
Nice, thank you.
I am able to run the GoXLR Interface, however. I cannot get hotkeys to work. Might be something I have missunderstood.

Where I am at now:

- Needed to use `npm install` before `npm run start`.
- I think the OBS Websocket address might be the [palakis plugin](https://github.com/Palakis/obs-websocket)?
- If I add a hotkey, it plays the sound, but nothing happen in the GoXLR App.
- The `GoXLR App` restarts on launch.

My setup is like this:
![image](https://user-images.githubusercontent.com/2109782/104842751-10e34c80-58c7-11eb-99f7-d09e057fe8c8.png)
![image](https://user-images.githubusercontent.com/2109782/104842833-a1ba2800-58c7-11eb-983f-f6593ecf41cd.png)

Also, it does not like when I used `ø` as a hotkey.
![image](https://user-images.githubusercontent.com/2109782/104842769-2fe1de80-58c7-11eb-8bf4-f6b42d010dd5.png)

**Comment by janecker** [comment]
You can ignore the OBS Websocket address for now. This will in future allow you to change the routing of your GoXLR when chaning the scenes in OBS Studio. It is kind of working but not through the GUI.

It seems to me that the GoXLR plugin is running, this is good. If the sound is playing, this also means that the hotkey assignment works. I suspect that the something with the Websocket connection between GoXLR Interface and the Stream Deck plugin is not working.

Do you by any chance also run the Elgato Stream Deck software? I am not sure if it will work if GoXLR Interface and the original Elgato Stream Deck software is running..

**Comment by janecker** [comment]
Another thought: I am seen that you also develop on a connector for some stream deck alternative. Was this also running while you where testing? I am not sure about the interface that is used between the Stream Deck plugin and the GoXLR application, maybe only it accepts one "connection"?

**Comment by oddbear** [comment]
I figured it out, it was some issues with the connection. ;)
Closing all applications, and reset the firewall rules worked.

**Comment by janecker** [comment]
Great to hear! Could I ask you to raise a new issue for the issue with your ø key? I thinks this is something worth fixing and I think we can close this issue. I may get in touch with you in future. I already saw your app when I first released my code and I was imidiatly thinking that your code may make it possible to no longer depend on the Stream Deck plugin.
